checker/types.go

```diff
@@ -489,3 +489,16 @@ func substitute(m *mapping, t *exprpb.Type, typeParamToDyn bool) *exprpb.Type {
 func typeKey(t *exprpb.Type) string {
 	return FormatCheckedType(t)
 }
+
+func mergeFunctionTypes(f *exprpb.Type_FunctionType) []*exprpb.Type {
```

Comment:
Consider naming as `flattenFunctionTypes()` and add the docstring

```
// flattenFunctionTypes takes a function with arg types T1, T2, ..., TN and result type TR
// and returns a slice containing {T1, T2, ..., TN, TR}.
```

Reply:
Done.